subsys/CMakeLists.txt

```
@@ -30,3 +30,4 @@ add_subdirectory(canbus)
 add_subdirectory_ifdef(CONFIG_TIMING_FUNCTIONS     timing)
 add_subdirectory_ifdef(CONFIG_DEMAND_PAGING        demand_paging)
 add_subdirectory(modbus)
+add_subdirectory(elfloader)
```

unless you are expecting more kconfig settings to enable elfloader, then I believe it's cleaner to ifdef the whole subdirectory, like this:
```suggestion
add_subdirectory(CONFIG_DYNAMIC_MODULE elfloader)
```